Assets/Scripts/Dialogue/InkManager.cs

```diff
@@ -0,0 +1,241 @@
+using Ink.Runtime;
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+using System;
+using System.Collections.Generic;
+
+public class InkManager : MonoBehaviour
+{
+    [SerializeField]
+    private TextAsset inkJSON;
+    [SerializeField]
+    private TextMeshProUGUI dialogueText;
+    [SerializeField]
+    private Button choiceButtonPrefab;
+    [SerializeField]
+    private Transform choiceButtonParent;
+    [SerializeField]
+    private Image playerImage, npcImage;
+    [SerializeField]
+    private List<CharacterSpriteSO> characterSprites = new List<CharacterSpriteSO>();
+
+    private Story story;
+    private bool isDialogueActive = false;
+
+    private const string TEST_TAG = "testTag";
+    private const string SPEAKER_TAG = "speaker";
+
+    private const string LOCKED_BTNTAG = "locked";
+
+    public static InkManager Instance { get; private set; }
+
+    public event Action OnDialogueStart;
+    public event Action OnDialogueEnd;
+
+    private void Awake()
+    {
+        if (Instance == null)
+            Instance = this;
+        else
+            Debug.LogError("Multiple instances of InkManager have been found", this);
```

Comment:
What happens right now is that it will throw an error if there are multiple instances found which will stop the game. So, I don't think it's necessary to then also delete the gameobject

Reply:
ah alright, didn't know that LogError stops the game, good to know